plugin-server/src/kafka/producer.ts

```diff
@@ -65,25 +65,44 @@ export const produce = async ({
     value,
     key,
     headers = [],
+    waitForAck = true,
 }: {
     producer: RdKafkaProducer
     topic: string
     value: Buffer | null
     key: Buffer | null
     headers?: { [key: string]: Buffer }[]
+    waitForAck?: boolean
```

Comment:
non-blocking

Rather than adding something to this interface, could we simply not await the promise in `KafkaProducerWrapper.queueMessage`?

Reply:
I still want to block and handle errors for `producer.produce` itself: one such error case is if the producer buffer is full.

We're leaving some perf on the table, but that's under-promising and over-delivering ^^